File: leet/0024-swap-nodes-in-pairs/swap-nodes-in-pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
    if head == nil || head.Next == nil {return head}
    
    var dummy ListNode
    dummy.Next = head
    
    last := &dummy
    current := dummy.Next
    next := current.Next
    
    for current != nil && next != nil{
        last.Next = next
        current.Next = next.Next
        next.Next = current
        
        last = current
        current = current.Next
        if current != nil {
            next = current.Next
        }
    }
    
    return dummy.Next
}

File: leet/0024-swap-nodes-in-pairs/swap-nodes-in-pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    let dummy = new ListNode()
    dummy.next = head
    
    let last = dummy
    
    while(last){
        let n1 = last.next
        if(!n1) break
        let n2 = n1.next
        if(!n2) break
        let next = n2.next
        
        last.next = n2
        n2.next = n1
        n1.next = next
        
        last = n1
    }
    
    return dummy.next
};
